### ISSUE TITLE
evaluator: conditional embedding of a definition incorrectly disallows fields

### ISSUE THREAD

**Issue by myitcv** [cue-lang/cue#1860]
Raising off the back of #1858. 

### What version of CUE are you using (`cue version`)?

<pre>
$ cue version
cue version v0.0.0-20220815093845-eab8c8ea5c60

       -compiler gc
     CGO_ENABLED 1
          GOARCH arm64
            GOOS linux
             vcs git
    vcs.revision eab8c8ea5c60ce284b4297ca707fa91cbce11b59
        vcs.time 2022-08-15T09:38:45Z
    vcs.modified false
</pre>

### Does this issue reproduce with the latest release?

Yes. This is also a bug with `v0.4.3`.

### What did you do?

```
exec cue eval x.cue
cmp stdout stdout.golden

-- x.cue --
package test

#Def1: {
	field1: string
}

#Def2: {
	cond: bool
	#Def1
	field2: string
}

#Def3: {
	cond: bool
	if cond {
		#Def1
	}
	field2: string
}

val1: #Def1 & {
	field1: "value1"
}

val2: #Def2 & {
	cond:   true
	field1: "value1"
	field2: "value2"
}

val3: #Def3 & {
	cond:   true
	field1: "value1"
	field2: "value2"
}
-- stdout.golden --
#Def1: {
    field1: string
}
#Def2: {
    cond:   bool
    field1: string
    field2: string
}
#Def3: {
    cond: bool
    if cond {
        #Def1
    }
    field2: string
}
val1: {
    field1: "value1"
}
val2: {
    cond:   true
    field1: "value1"
    field2: "value2"
}
val3: {
    cond:   true
    field1: "value1"
    field2: "value2"
}
```

### What did you expect to see?

Passing test.

### What did you see instead?

```
> exec cue eval x.cue
[stderr]
val3: field cond not allowed:
    ./x.cue:3:8
    ./x.cue:14:2
    ./x.cue:15:2
    ./x.cue:15:10
    ./x.cue:16:3
    ./x.cue:31:7
    ./x.cue:32:2
val3: field field2 not allowed:
    ./x.cue:3:8
    ./x.cue:15:2
    ./x.cue:15:10
    ./x.cue:16:3
    ./x.cue:18:2
    ./x.cue:31:7
    ./x.cue:34:2
[exit status 1]
FAIL: /tmp/testscript1789739620/repro.txtar/script.txt:1: unexpected command failure
```

Thanks to @yann-soubeyrand for raising.

**Comment by rogpeppe** [comment]
Here's a slightly simpler case that I think exhibits the same issue:

```
exec cue export tst.cue
cmp stdout golden.cue

-- tst.cue --
#Def1: field1: string

#Def2: {
	if true {
		#Def1
	}
	field2: string
}

val: #Def2 & {
	field1: "value1"
	field2: "value2"
}
-- golden.cue --
{
    "val": {
        "field1": "value1",
        "field2": "value2"
    }
}
```

It fails with:

```
#Def2: field not allowed: field2:
    ./tst.cue:1:8
    ./tst.cue:4:2
    ./tst.cue:4:10
    ./tst.cue:5:3
    ./tst.cue:7:2
val: field not allowed: field2:
    ./tst.cue:1:8
    ./tst.cue:4:2
    ./tst.cue:4:10
    ./tst.cue:5:3
    ./tst.cue:7:2
    ./tst.cue:10:6
    ./tst.cue:12:2
[exit status 1]
```

**Comment by myitcv** [comment]
Apologies - the original report here lacked lots of information. Now corrected, and I've also confirmed this was fixed in ed4f8850d35a6ca4012c16db0cbf42b6e2d52af0.
